pelican/plugins/more_categories/more_categories.py

```diff
@@ -75,9 +74,24 @@ def create_categories(generator):
 
     generator.categories = sorted(
         list(cat_dct.items()),
-        reverse=generator.settings.get('REVERSE_CATEGORY_ORDER') or False,
+        reverse=generator.settings.get("REVERSE_CATEGORY_ORDER") or False,
     )
-    generator._update_context(['categories'])
+    generator._update_context(["categories"])
+
+    # Add subcategories
+    cats = {}
+    for category, articles in generator.categories:
+        for anc in category.ancestors:
+            if anc != category:
+                if anc.slug in cats:
+                    cats[anc.slug].append(category)
+                else:
+                    cats[anc.slug] = [category]
```

Comment:
You can simply use the category itself as the key, since its hash is the hash of its slug.